src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.tsx

```diff
@@ -4,61 +4,150 @@ import { connect } from 'react-redux';
 
 import { RootState } from 'src/store';
 import { EnsObject } from 'src/ens-object/ensObjectTypes';
-import { GenomeInfoData } from 'src/genome/genomeTypes';
 
 import { getBrowserActiveGenomeId } from '../../../browserSelectors';
-import { getGenomeInfo } from 'src/genome/genomeSelectors';
+import { updateTrackStates } from 'src/content/app/browser/browserActions';
+import { TrackStates } from 'src/content/app/browser/track-panel/trackPanelConfig';
+import { getActiveGenomePreviouslyViewedObjects } from 'src/content/app/browser/track-panel/trackPanelSelectors';
 import { fetchExampleEnsObjects } from 'src/ens-object/ensObjectActions';
 import { getExampleEnsObjects } from 'src/ens-object/ensObjectSelectors';
 import * as urlFor from 'src/shared/helpers/urlHelper';
 import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
+import { closeTrackPanelModal } from '../../trackPanelActions';
 
 import upperFirst from 'lodash/upperFirst';
 
 import styles from '../TrackPanelModal.scss';
+import { Bookmark } from '../../trackPanelState';
 
-type TrackPanelBookmarksProps = {
+type StateProps = {
   activeGenomeId: string | null;
-  genomeInfo: GenomeInfoData;
   exampleEnsObjects: EnsObject[];
+  previouslyViewedObjects: Bookmark[];
+};
+
+type DispatchProps = {
   fetchExampleEnsObjects: (objectId: string) => void;
+  updateTrackStates: (trackStates: TrackStates) => void;
+  closeTrackPanelModal: () => void;
+};
+export type TrackPanelBookmarksProps = StateProps & DispatchProps;
+
+const getExampleObjLabel = (exampleObject: EnsObject | Bookmark) => {
```

Comment:
1) name of the function no longer describes what the function does
2) both gene and region objects now have the `label` field
3) is this function even necessary anymore?

Reply:
Not sure of label was available for reigns when I wrote this.
Removed this function now.